app/helpers/markdown_form_builder.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+class MarkdownFormBuilder < ActionView::Helpers::FormBuilder
+  def markdown(method, args = {})
+    # Adopt simple form builder to work with form_for
+    @attribute_name = method
+    @input_html_options = args[:input_html]
+
+    @template.capture do
+      @template.concat form_textarea
+      @template.concat toast_ui_editor
+    end
+  end
+
+  private
+
+  def form_textarea
+    @template.text_area @object_name, @attribute_name,
+    **(@input_html_options || {}),
+      id: label_target,
+      class: 'd-none'
+  end
+
+  def toast_ui_editor
+    @template.tag.div(class: 'markdown-editor__wrapper') do
+      @template.concat @template.tag.div(class: 'markdown-editor', data: {behavior: 'markdown-editor-widget', id: label_target})
+      @template.concat resize_btn
+    end
+  end
+
+  def resize_btn
+    @template.tag.button(class: 'markdown-editor__resize-btn fa-solid', type: 'button', id: "#{label_target}-resize",
+      title: I18n.t(:'markdown_editor.expand'), aria_label: I18n.t(:'markdown_editor.expand'))
+  end
```

Comment:
(This is probably not the right location, too. And, I am really sorry that I discovered another issue 🙈, this should not touch the great results you achieved way faster than I imagined in any way!)

This new resize button works like a charm. There is just a minor visual issue I noticed: On Xikolo, the button is always in the corner, independent of the text:

<img width="771" alt="Bildschirmfoto 2024-04-16 um 00 58 59" src="https://github.com/openHPI/codeocean/assets/7300329/a8e42aa9-c42e-43b9-8ce5-a63b8470d2f8">

On CodeOcean, however, the button is only in the corner when the editor is fully expanded:

<img width="1339" alt="Bildschirmfoto 2024-04-16 um 00 59 16" src="https://github.com/openHPI/codeocean/assets/7300329/09aaa37a-1787-4ad1-8e80-68d18e611843">

Otherwise, the button has some spacing to the right corner, but not below:

<img width="1315" alt="Bildschirmfoto 2024-04-16 um 00 59 06" src="https://github.com/openHPI/codeocean/assets/7300329/2eb267be-37bb-46b3-9a29-589dd1b41562">

This positional change is especially noteworthy when switching back from expanded mode to "collapsed" mode.

Reply:
No problem! Thank you for the extensive reviews, that's the way to improve things ;)

What's happening here is that this is designed taking into account the scroll-bars and assuming they are visible. Usually that's the behaviour when using an external mouse (or when [setting the options to always do so](https://support.apple.com/guide/mac-help/change-appearance-settings-mchlp1225/mac#:~:text=Automatically%20based%20on%20mouse%20or,bars%20until%20you%20start%20scrolling)). In the expanded state the button is in the corner because they are never there, and in the collapsed state it depends on the content. Is it possible that when you took this screenshot you were using your mouse-track?

What's confusing me a bit is the screenshot from Xikolo, this should work the same way there too. Is this a course in production I can go check?

Reply:
I was taking both screenshots at the same condition mouse-wise (no mouse connected, only my trackpad, scrollbars set to "auto" in macOS settings). The Xikolo screenshot is taken from this [course forum](https://open.hpi.de/courses/javaeinstieg-schule2024/pinboard) after "Start a new topic" was clicked.

Reply:
I just noticed that the behavior on Xikolo is indeed different. For a text item (e.g., [this one](https://open.hpi.de/courses/pythonjunior-schule2024/sections/YaJ3MVmrE1NsZVYUGOmeZ/items/5iKFW4Ef9znO9vKHi2yfiY/edit)), the expand button is also outside the scrollbars. Still, on the same platform in the [course forum](https://open.hpi.de/courses/javaeinstieg-schule2024/pinboard), the expand button is overlapping with the scrollbars.

Reply:
I've opened a MR in xikolo: https://lab.xikolo.de/xikolo/web/-/merge_requests/4494 and also adjusted the code here.

The behaviour should be as described earlier: In the expanded state the button is in the corner because no scrollbar is displayed and in the collapsed state it depends on the content. If the user is using a trackpad it might feel a bit misplaced when not scrolling, but otherwise the button would appear on top of the scrollbar.

Reply:
Alright, thank you! I was now able to confirm this behavior: When scrolling becomes possible (due to the text length), the button moves away from the scrollbars (i.e., to the left). When the text becomes shorter, the button moves to the corner again. This is consistent and works as expected :+1:

<img width="235" alt="Bildschirmfoto 2024-04-17 um 18 05 04" src="https://github.com/openHPI/codeocean/assets/7300329/e6bd22b3-386c-419a-8e94-cd0ab3d742ed">